CosmoTech_Acceleration_Library/Accelerators/scenario_download/scenario_downloader.py

```diff
@@ -50,27 +52,88 @@ def download_dataset(self, dataset_id: str) -> (str, str, Union[str, None]):
         with cosmotech_api.ApiClient(self.configuration) as api_client:
             api_instance = DatasetApi(api_client)
 
-            dataset = api_instance.find_dataset_by_id(organization_id=self.organization_id,
-                                                      dataset_id=dataset_id)
+            dataset = api_instance.find_dataset_by_id(
+                organization_id=self.organization_id,
+                dataset_id=dataset_id)
+            parameters = dataset['connector']['parameters_values']
 
-            is_adt = 'AZURE_DIGITAL_TWINS_URL' in dataset['connector']['parameters_values']
+            is_adt = 'AZURE_DIGITAL_TWINS_URL' in parameters
+            is_twin_cache = 'TWIN_CACHE_HOST' in parameters
 
             if is_adt:
-                return {"type": 'adt',
-                        "content": self._download_adt_content(adt_adress=dataset['connector']['parameters_values'][
-                            'AZURE_DIGITAL_TWINS_URL']), "name": dataset['name']}
+                return {
+                    "type": 'adt',
+                    "content": self._download_adt_content(
+                        adt_adress=parameters['AZURE_DIGITAL_TWINS_URL']),
+                    "name": dataset['name']}
+            elif is_twin_cache:
+                twin_cache_name = parameters['TWIN_CACHE_NAME']
+                return {
+                    "type": "adt",
```

Comment:
Type is `adt` and not twin cache, was it done on purpose ?

Reply:
I think @vysahjk left "adt" to avoid impacting the azure function code. Since the objects returned by the library have the same structure as previously with adt alone, he kept "adt". Yes, we could have changed for a more generic term in both repo